### PR TITLE
added func. to upload/query and download files

### PR DESCRIPTION
**Pull request by a96lex**
@FeaR6077 's contribution. Opened pull request to class-methods instead of main branch

### CODE REVIEW

File: aflr/api_resources/file.py [hunk 1]
```diff
@@ -1,19 +1,62 @@
+from posixpath import basename
 import aflr
-from aflr.api_request import APIRequest
+from aflr.helper_classes import ListableResource
+from tkinter import Tk
+from tkinter.filedialog import askopenfilename
```
Comment (a96lex) [inline aflr/api_resources/file.py:5]: I wouldn't add this package. Instead we can have an argument for `file_path`. If users want to visually choose a file, they can use tkinter from their scripts. this means we also don't need basename

you can handle this with `open(path, "rb")`

File: aflr/api_resources/file.py [hunk 2]
```diff
@@ -1,19 +1,62 @@
...
+        response = cls._get_request(path_param=cls.by_org)
+        return response
+
+    @classmethod
+    def search_audiofiles_by_tag(cls, tags):
+        url = cls._get_request(path_param=cls.by_tag + tags)
+        return url
+
+
+# get speech files
```
Comment (a96lex) [inline aflr/api_resources/file.py:56]: you can remove this

File: aflr/api_resources/file.py [hunk 3]
```diff
@@ -1,19 +1,62 @@
...
-    # TBD
+        # Do put request with presigned URL
+        response = requests.request(
+            "PUT", url=fileUploadUrl, headers=headers, data=payload
+        )
 
-    # get sound files
-    # TBD
+        response = {
+            "message": f" Success. Please make sure to save this mediaId : {mediaId}"
```
Comment (a96lex) [inline aflr/api_resources/file.py:36]: remove space f" S

File: aflr/api_resources/file.py [hunk 4]
```diff
@@ -1,19 +1,60 @@
...
+        }
+        return response
+
+    @classmethod
+    def download_audiofile(cls, metaId):
+        url = cls._get_request(path_param=cls.by_metaId + metaId)
+        return url
+
+    @classmethod
+    def search_audiofiles(cls):
```
Comment (a96lex) [inline aflr/api_resources/file.py:46]: pls merge this to `search_audiofiles_by_tag`

File: aflr/api_resources/file.py [hunk 5]
```diff
@@ -1,19 +1,60 @@
...
 
-    # get sound files
-    # TBD
+        response = {
+            "message": f" Success. Please make sure to save this mediaId : {mediaId}"
+        }
+        return response
+
+    @classmethod
+    def download_audiofile(cls, metaId):
```
Comment (a96lex) [inline aflr/api_resources/file.py:41]: rename to `get_download_url` or similar

File: aflr/api_resources/file.py [hunk 6]
```diff
@@ -1,19 +1,60 @@
...
+import os
+import requests
+import json
 
 
-class File(APIRequest):
+class File(ListableResource):
     OBJECT_NAME = "file"
+    resource_path = "/file"
+    custom_audio_upload = "/file/customaudio/uploadurl?filename="
```
Comment (a96lex) [inline aflr/api_resources/file.py:14]: i'd have only `custom_audio_resource_path = "file/customaudio"` and leave the rest to the methods

File: aflr/api_resources/file.py [hunk 7]
```diff
@@ -1,19 +1,60 @@
+from posixpath import basename
 import aflr
-from aflr.api_request import APIRequest
+from aflr.helper_classes import ListableResource
+from tkinter import Tk
```
Comment (a96lex) [inline aflr/api_resources/file.py:4]: what is this for? did not find it in the code

File: aflr/api_resources/file.py [hunk 8]
```diff
@@ -1,19 +1,60 @@
...
 
-    def config_test(self):
-        return f"Configured to transact {self.OBJECT_NAME} objects to {self.url} with api_key = {self.api_key}"
+        mediaId = url["mediaId"]
+        fileUploadUrl = url["fileUploadUrl"]
 
-    # get speech files
-    # TBD
+        # Do put request with presigned URL
+        response = requests.request(
```
Comment (a96lex) [inline aflr/api_resources/file.py:31]: this needs more work. we need a `._put_request()` method in the `APIResource` class. So there is no need to import requests here

Comment (a96lex) [reply aflr/api_resources/file.py:31]: you should call it as `cls._put_request()`

File: aflr/api_resources/file.py [hunk 9]
```diff
@@ -1,19 +1,60 @@
...
+from aflr.helper_classes import ListableResource
+from tkinter import Tk
+from tkinter.filedialog import askopenfilename
+import os
+import requests
+import json
 
 
-class File(APIRequest):
+class File(ListableResource):
```
Comment (a96lex) [inline aflr/api_resources/file.py:11]: you can inherit only from `APIRequest if we don't need `.list()` method

**Review by a96lex — COMMENTED** [review]
I can't request changes since I created this pr :( but please see my comments

File: aflr/api_resources/file.py (hunk 2 above)
Comment (a96lex) [reply aflr/api_resources/file.py:56]: all comments I mean

Comment (FeaR6077) [reply aflr/api_resources/file.py:56]: done

File: aflr/api_resources/file.py (hunk 3 above)
Comment (FeaR6077) [reply aflr/api_resources/file.py:36]: done

File: aflr/api_resources/file.py (hunk 6 above)
Comment (FeaR6077) [reply aflr/api_resources/file.py:14]: done

File: aflr/api_resources/file.py (hunk 7 above)
Comment (FeaR6077) [reply aflr/api_resources/file.py:4]: deleted

File: aflr/api_resources/file.py (hunk 8 above)
Comment (FeaR6077) [reply aflr/api_resources/file.py:31]: done

File: aflr/api_resources/file.py (hunk 1 above)
Comment (FeaR6077) [reply aflr/api_resources/file.py:5]: done

File: aflr/api_resources/file.py (hunk 4 above)
Comment (FeaR6077) [reply aflr/api_resources/file.py:46]: done

**Comment by FeaR6077** [comment]
all changes done to this pr:
https://github.com/aflorithmic/aflr_python/pull/34

**Pull request closed without merging**